lib/model/store.dart

```diff
@@ -893,15 +896,31 @@ class UpdateMachine {
     store.updateMachine = this;
   }
 
-  /// Load the user's data from the server, and start an event queue going.
+  /// Load data for the given account ID from the server,
+  /// and start an event queue going.
+  ///
+  /// The account for `accountId` must exist.
+  ///
+  /// Throws [AccountNotFoundException] after an async gap
+  /// if the account has been removed.
   ///
   /// In the future this might load an old snapshot from local storage first.
   static Future<UpdateMachine> load(GlobalStore globalStore, int accountId) async {
     Account account = globalStore.getAccount(accountId)!;
     final connection = globalStore.apiConnectionFromAccount(account);
 
+    void stopAndThrowIfNoAccount() {
+      final account = globalStore.getAccount(accountId);
+      if (account == null) {
+        assert(debugLog('Account logged out during UpdateMachine.load'));
+        connection.close();
+        throw AccountNotFoundException();
+      }
+    }
+
     final stopwatch = Stopwatch()..start();
-    final initialSnapshot = await _registerQueueWithRetry(connection);
+    final initialSnapshot = await _registerQueueWithRetry(connection,
+      stopAndThrowIfNoAccount: stopAndThrowIfNoAccount);
     final t = (stopwatch..stop()).elapsed;
     assert(debugLog("initial fetch time: ${t.inMilliseconds}ms"));
 
```

Comment:
Further down there is another async gap at `await globalStore.updateAccount`; should we call `stopAndThrowIfNoAccount` there as well?

Reply:
I believe there isn't a bug that we could catch by doing so.
1. We hope and expect that `await globalStore.updateAccount` always fails if the account is removed concurrently. If it doesn't, that's a bug in the database implementation.
2. If `await globalStore.updateAccount` succeeds, that means there's still an `Account` object in `GlobalStore._accounts`. Since `stopAndThrowIfNoAccount` looks at `GlobalStore._accounts` too, it would find it there and wouldn't throw. So even if the (1) bug exists, `stopAndThrowIfNoAccount` wouldn't help us catch it.